DOC: Add rolling in reference/groupby.rst

luke396 opened the pull request:
- [ ] close #51097

Commit:
Add rolling in reference/groupby.rst

mroeschke commented:
Since we have docstring checks, the docstrings of these will need to be improved in this PR

```
Error: /home/runner/work/pandas/pandas/pandas/core/groupby/groupby.py:2787:RT01:pandas.core.groupby.DataFrameGroupBy.rolling:No Returns section found
Error: /home/runner/work/pandas/pandas/pandas/core/groupby/groupby.py:2787:RT01:pandas.core.groupby.SeriesGroupBy.rolling:No Returns section found
Error: /home/runner/work/pandas/pandas/pandas/core/groupby/groupby.py:2787:EX01:pandas.core.groupby.DataFrameGroupBy.rolling:No examples section found
Error: /home/runner/work/pandas/pandas/pandas/core/groupby/groupby.py:2787:EX01:pandas.core.groupby.SeriesGroupBy.rolling:No examples section found
```

Commit:
Add more docstring

luke396 commented:
> Since we have docstring checks, the docstrings of these will need to be improved in this PR
> 
> ```
> Error: /home/runner/work/pandas/pandas/pandas/core/groupby/groupby.py:2787:RT01:pandas.core.groupby.DataFrameGroupBy.rolling:No Returns section found
> Error: /home/runner/work/pandas/pandas/pandas/core/groupby/groupby.py:2787:RT01:pandas.core.groupby.SeriesGroupBy.rolling:No Returns section found
> Error: /home/runner/work/pandas/pandas/pandas/core/groupby/groupby.py:2787:EX01:pandas.core.groupby.DataFrameGroupBy.rolling:No examples section found
> Error: /home/runner/work/pandas/pandas/pandas/core/groupby/groupby.py:2787:EX01:pandas.core.groupby.SeriesGroupBy.rolling:No examples section found
> ```

I added some docstring, but they definitely need further discussion. 

The parameters section is copied from [pandas.DataFrame.rolling](https://pandas.pydata.org/docs/dev/reference/api/pandas.DataFrame.rolling.html#pandas.DataFrame.rolling), the parameter `step` is not implemented, so I removed it.  Other parameters are implemented, but the exact description I'm not sure if I need to keep changing.

Commit:
Add see also

Commit:
Change See also

Commit:
Fix ci

Commit:
Fix ci

luke396 committed:
Merge branch 'main' into add-doc-groupy.rolling

mroeschke commented on pandas/core/groupby/groupby.py:
```diff
@@ -2723,10 +2723,134 @@ def resample(self, rule, *args, **kwargs):
+        2     2  3  1.267
+        3     2  4 -0.562
+
+        >>> df.groupby('A').rolling(2).sum()
```
These examples should show calling groupby rolling specifically

luke396 replied:
Sorry, I didn't fully understand what changes should be made to it. Do you mean that these examples do not show the usefulness of groupy rolling and that the data should be reorganized?

mroeschke replied:
Ah sorry my mistake. The examples look good

mroeschke commented on pandas/core/groupby/groupby.py:
```diff
@@ -2723,10 +2723,134 @@ def resample(self, rule, *args, **kwargs):
 
     @final
     @Substitution(name="groupby")
```
I think this can be removed too

luke396 replied:
Fine, how about this time?

Commit:
Remove substituion

mroeschke commented:
Thanks @luke396
